src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dataParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const dateObject = {};
+  let result = '';
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateObject[fromFormat[i]] = dataParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObject['YY'] = dateObject['YYYY'].slice(2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const yy = dateObject['YY'];
+
+    if (parseInt(yy, 10) < 30) {
+      dateObject['YYYY'] = `20${yy}`;
+    }
+
+    if (parseInt(yy, 10) >= 30) {
+      dateObject['YYYY'] = `19${yy}`;
+    }
+  }
+
+  const resultParts = toFormat.slice(0, -1).map((part) => dateObject[part]);
```

Comment:
Similar to the issue with `fromFormat`, the assumption that the last element of `toFormat` is the separator might not hold true. Ensure that the separator is correctly identified or structured in the format array.